reference/conanfile/tools/meson/mesontoolchain.rst

```diff
@@ -116,6 +116,50 @@ The ``MesonToolchain`` only works with the ``PkgConfigDeps`` generator.
 Please, do not use other generators, as they can have overlapping definitions that can conflict.
 
 
+Default directories
++++++++++++++++++++++
+
+Since Conan 1.51, ``MesonToolchain`` manages some of the directories used by Meson. They are `conan_meson_native.ini`
+and `conan_meson_cross.ini` variables declared under the ``[project options]`` section
+(see more information about `Meson directories <https://mesonbuild.com/Builtin-options.html#directories>`__):
+
+
+``bindir``: value coming from ``self.cpp.package.bindirs``. Defaulted to None.
+``sbindir``: value coming from ``self.cpp.package.bindirs``. Defaulted to None.
+``libexecdir``: value coming from ``self.cpp.package.bindirs``. Defaulted to None.
+``datadir``: value coming from ``self.cpp.package.resdirs``. Defaulted to None.
+``localedir``: value coming from ``self.cpp.package.resdirs``. Defaulted to None.
+``mandir``: value coming from ``self.cpp.package.resdirs``. Defaulted to None.
+``infodir``: value coming from ``self.cpp.package.resdirs``. Defaulted to None.
+``includedir``: value coming from ``self.cpp.package.includedirs``. Defaulted to None.
+``libdir``: value coming from ``self.cpp.package.libdirs``. Defaulted to None.
+
+Notice that it needs a ``layout`` to be able to initialize those ``self.cpp.package.xxxxx`` variables. For instance:
+
+.. code:: python
+
+    from conan import ConanFile
+    from conan.tools.meson import MesonToolchain
+
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+
+        def layout(self):
+            self.folders.build = "build"
+            self.cpp.package.resdirs = ["res"]
+
+        def generate(self):
+            tc = MesonToolchain(self)
+            print(tc.project_options["datadir"])  # Will print '["res"]'
```

Comment:
In general it is better to use ``self.output.info()`` etc, because this output will be automatically prefixed with the package scope.